pkg/kn/commands/service/export_test.go

```diff
@@ -0,0 +1,322 @@
+	template.Spec.Containers[0].Image = "gcr.io/foo/bar:baz"
+	svcs = append(svcs, labelService)
+
+	//case 4 - config map
+	CMservice := getService("foo")
+	template = &CMservice.Spec.Template
+	template.Spec.Containers[0].EnvFrom = []v1.EnvFromSource{
+		{
+			ConfigMapRef: &v1.ConfigMapEnvSource{
+				LocalObjectReference: v1.LocalObjectReference{
+					Name: "config-map-name",
+				},
+			},
+		},
+	}
+	template.Spec.Containers[0].Image = "gcr.io/foo/bar:baz"
+	template.Annotations = map[string]string{servinglib.UserImageAnnotationKey: "gcr.io/foo/bar:baz"}
+	svcs = append(svcs, CMservice)
+
+	//case 5 - volume mount and secrets
+	Volservice := getService("foo")
+	template = &Volservice.Spec.Template
+	template.Spec.Volumes = []v1.Volume{
+		{
+			Name: "volume-name",
+			VolumeSource: v1.VolumeSource{
+				Secret: &v1.SecretVolumeSource{
+					SecretName: "secret-name",
+				},
+			},
+		},
+	}
+
+	template.Spec.Containers[0].VolumeMounts = []v1.VolumeMount{
+		{
+			Name:      "volume-name",
+			MountPath: "/mount/path",
+			ReadOnly:  true,
+		},
+	}
+	svcs = append(svcs, Volservice)
+
+	template.Spec.Containers[0].Image = "gcr.io/foo/bar:baz"
+	template.Annotations = map[string]string{servinglib.UserImageAnnotationKey: "gcr.io/foo/bar:baz"}
+
+	for _, svc := range svcs {
+		svc.TypeMeta = typeMeta
+		callServiceExportTest(t, svc)
+	}
+
+}
+
+func callServiceExportTest(t *testing.T, expectedService *servingv1.Service) {
+	// New mock client
+	client := knclient.NewMockKnServiceClient(t)
+
+	// Recording:
+	r := client.Recorder()
+
+	r.GetService(expectedService.ObjectMeta.Name, expectedService, nil)
+
+	output, err := executeServiceCommand(client, "export", expectedService.ObjectMeta.Name, "-o", "json")
+
+	assert.NilError(t, err)
+
+	stripExpectedSvcVariables(expectedService)
+
+	actSvc := servingv1.Service{}
+
+	json.Unmarshal([]byte(output), &actSvc)
+
+	assert.DeepEqual(t, expectedService, &actSvc)
+
+	// Validate that all recorded API methods have been called
+	r.Validate()
+}
+
+func TestServiceExportwithMultipleRevisions(t *testing.T) {
+
+	//case 1 = 2 revisions with traffic split
+	trafficSplitService := createServiceTwoRevsionsWithTraffic("foo", true)
+
+	multiRevs := createTestRevisionList("rev", "foo")
+
+	callServiceExportHistoryTest(t, trafficSplitService, multiRevs)
+
+	//case 2 - same revsions no traffic split
+	noTrafficSplitService := createServiceTwoRevsionsWithTraffic("foo", false)
+
+	callServiceExportHistoryTest(t, noTrafficSplitService, multiRevs)
+}
+
+func callServiceExportHistoryTest(t *testing.T, expectedService *servingv1.Service, revs *servingv1.RevisionList) {
+	// New mock client
+	client := knclient.NewMockKnServiceClient(t)
+
+	// Recording:
+	r := client.Recorder()
+
+	r.GetService(expectedService.ObjectMeta.Name, expectedService, nil)
+
+	r.ListRevisions(mock.Any(), revs, nil)
+
+	output, err := executeServiceCommand(client, "export", expectedService.ObjectMeta.Name, "-r", "-o", "json")
```

Comment:
you should change this to generate YAML and verify. The one above does JSON as well. Seems redundant to test that output format twice.

Reply:
have added another test for yaml